docs/sdk-and-tools/rest-api/transactions.md

```diff
@@ -479,3 +515,249 @@ Transaction status retrieved successfully.
 :::important
 The optional query parameter **`sender`** is only applicable to requests against the Proxy (not against the Observer Nodes).
 :::
+
+## <span class="badge badge-primary">GET</span> **Get Transactions Pool**
+
+`http://local-proxy-instance/transaction/pool`
+
+:::warning
+This endpoint isn't available on public gateway. However, it can be used on a local proxy instance, by setting `AllowEntireTxPoolFetch` to `true`
+:::
+
+This endpoint allows one to fetch the entire transactions pool, merging the pools from each shard.
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--Request-->
+
+<!--Response-->
+
+🟢 200: OK
+
+Transaction status retrieved successfully.
+
+```
+{
+    "data": {
+        "txPool": {
+            "regularTransactions": [
+                {
+                    "txFields": {
+                        "gasLimit": 10,
+                        "gasPrice": 1000,
+                        "receiver": "erd1...",
+                        "sender": "erd1...",
+                        "value": "10000000000000000000"
+                    }
+                }
+            ],
+            "smartContractResults": [
+                {
+                    "txFields": {
+                        "gasLimit": 10,
+                        "gasPrice": 1000,
+                        "receiver": "erd1...",
+                        "sender": "erd1...",
+                        "value": "10000000000000000000"
+                    }
+                }
+            ],
+            "rewards": [
+                {
+                    "txFields": {
+                        "gasLimit": 10,
+                        "gasPrice": 1000,
+                        "receiver": "erd1...",
+                        "sender": "erd1...",
+                        "value": "10000000000000000000"
+                    }
+                }
+            ]
+        }
+    },
+    "error": "",
+    "code": "successful"
+}
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## <span class="badge badge-primary">GET</span> **Get Transactions Pool for a Sender**
+
+`https://gateway.elrond.com/transaction/pool?by-sender=:sender:`
+
+This endpoint allows one to fetch all the transactions of a sender from the transactions pool.
+
+### Example with no field
```

Comment:
I would duplicate L591-L682 inside `Get Transactions Pool` as well as fields apply in both cases

Reply:
done